### PR TITLE
nengo-gui: init at 0.4.8

### PR DESCRIPTION
**Pull request by arjix**
###### Motivation for this change
Add nengo-gui to python packages

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
there should be a entry in `pkgs/top-level/python-packages.nix` so that it's been added to the ptyhon package sets

File: pkgs/development/python-modules/nengo-gui/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib, fetchFromGitHub, buildPythonPackage,
+  nengo,
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nengo-gui/default.nix:2]:
```suggestion
{ lib, fetchFromGitHub, buildPythonPackage
, nengo
```

File: pkgs/development/python-modules/nengo-gui/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Nengo interactive visualizer";
+    homepage    = "https://nengo.ai/";
+    license     = licenses.unfreeRedistributable;
+    maintainers = with maintainers; [ arjix ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nengo-gui/default.nix:29]: One final new line to much.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -4726,6 +4726,8 @@ in {
 
   ndtypes = callPackage ../development/python-modules/ndtypes { };
 
+  nengo-gui = callPackage ../development/python-modules/nengo-gui { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:4729]: This should be moved out of pythonPackages.

Comment (arjix) [reply pkgs/top-level/python-packages.nix:4729]: moved where to ?

File: pkgs/development/python-modules/nengo-gui/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{ lib, fetchFromGitHub, buildPythonPackage,
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/nengo-gui/default.nix:1]:
```suggestion
{ lib, fetchFromGitHub, buildPythonPackage
```

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/top-level/python-packages.nix:4729]: top-level packages because it is not a python library but an end user application, right?

Comment (arjix) [reply pkgs/top-level/python-packages.nix:4729]: it's a front end to a python library with a built-in editor, would applications/science/machine-learning or applications/editors be better suited for that ?

Comment (SuperSandro2000) [reply pkgs/top-level/python-packages.nix:4729]: > applications/editors

that would work for me.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -29923,6 +29923,8 @@ in
 
   ### SCIENCE/MACHINE LEARNING
 
+  nengo-gui = python3Packages.callPackage ../applications/science/machine-learning/nengo-gui { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:29926]:
```suggestion
  nengo-gui = callPackage ../applications/science/machine-learning/nengo-gui { };
```

File: pkgs/applications/science/machine-learning/nengo-gui/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{ lib, fetchFromGitHub, buildPythonPackage
+, nengo
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/nengo-gui/default.nix:2]:
```suggestion
{ lib, fetchFromGitHub, python3Packages
```

File: pkgs/applications/science/machine-learning/nengo-gui/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+}:
+
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/nengo-gui/default.nix:5]:
```suggestion
python3Packages.buildPythonPackage rec {
```

File: pkgs/applications/science/machine-learning/nengo-gui/default.nix [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  version = "0.4.8";
+
+  src = fetchFromGitHub {
+    owner = "nengo";
+    repo = "nengo-gui";
+    rev = "v${version}";
+    sha256 = "1awb0h2l6yifb77zah7a4qzxqvkk4ac5fynangalidr10sk9rzk3";
+  };
+
+  propagatedBuildInputs = [ nengo ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/nengo-gui/default.nix:16]:
```suggestion
  propagatedBuildInputs = with python3Packages; [ nengo ];
```

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:29926]: Why can't I apply this?

File: pkgs/applications/science/machine-learning/nengo-gui/default.nix [hunk 9]
```diff
@@ -0,0 +1,27 @@
+{ lib, fetchFromGitHub, python3Packages
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/nengo-gui/default.nix:2]:
```suggestion
{ lib, fetchFromGitHub, python3Packages }:
```

**Pull request merged**